src/img_to_text_gen.py

```diff
@@ -0,0 +1,50 @@
+import torch
+from transformers import VisionEncoderDecoderModel, \
+    ViTFeatureExtractor, AutoTokenizer
+
+
+class IttGenerator(object):
+    def __new__(cls):
+        if not hasattr(cls, 'instance'):
+            cls.instance = super(IttGenerator, cls).__new__(cls)
+        return cls.instance
+
+    model = VisionEncoderDecoderModel\
+        .from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+    feature_extractor = ViTFeatureExtractor\
+        .from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+    tokenizer = AutoTokenizer\
+        .from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+    device = torch\
+        .device("cuda" if torch.cuda.is_available() else "cpu")
+    model.to(device)

```

Comment:
Put these lines under `def __init__` method to ensure that all loading happened only on class initialization.